avocado/utils/memory.py

```diff
@@ -48,6 +48,25 @@ def memtotal():
     return read_from_meminfo('MemTotal')
 
 
+def memtotal_sys():
+    """
+    Reports actual memory size according to online-memory
+    blocks available via "/sys"
+
+    :return: system memory in Kb as float
+    """
+    sys_mempath = '/sys/devices/system/memory'
+    no_memblocks = 0
+    for directory in os.listdir(sys_mempath):
```

Comment:
yep, the use of `glob.glob("/sys/devices/system/memory/memory*")` would be probably better, but this function is not likely to be called too often 😄.